sled-agent/src/sled_agent.rs

```diff
@@ -251,26 +251,22 @@ impl SledAgent {
                 storage.upsert_zpool(pool.clone(), None).await;
             }
         }
+        let hardware = HardwareManager::new(&parent_log, services.sled_mode())
+            .map_err(|e| Error::Hardware(e))?;
+
         let instances = InstanceManager::new(
             parent_log.clone(),
             lazy_nexus_client.clone(),
             etherstub.clone(),
             *sled_address.ip(),
             request.gateway.mac.0,
         )?;
+        instances.set_reservoir_size(&hardware)?;
```

Comment:
Do we anticipate changing the reservoir size at runtime? If not maybe just set it as part of `InstanceManager::new`. Would also let you get rid of the `Mutex`

Reply:
I actually intentionally kept it separate -- I think that in the future, we may want Nexus to have control over this value, and actually modify it at runtime. It seems like a powerful tool for binpacking purposes to be able to tweak this on a per-sled basis.

(Namely, the "80% of Phys RAM goes to the reservoir" bit is totally arbitrary, but we could make this smaller or larger on sleds dedicated more or less towards guest workloads)